web-server/v0.4/src/components/TableTree/index.js

```diff
@@ -0,0 +1,111 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'antd';
+
+export default class TableTree extends PureComponent {
+  static propTypes = {
+    id: PropTypes.string,
+    dataSource: PropTypes.array,
+    extension: PropTypes.array,
+    onload: PropTypes.func,
+    config: PropTypes.object,
+  };
+
+  static defaultProps = {
+    id: '',
+    dataSource: [],
+    extension: [],
+    onload: () => {},
+    config: {},
+  };
+
+  constructor(props) {
+    super(props);
+    this.fileData = [];
+    this.state = {
+      expandedKeys: [],
+    };
+  }
+
+  componentDidMount() {
+    const { dataSource, onload } = this.props;
+    onload(dataSource);
```

Comment:
We should consider referencing table data for `TableTree` as a component prop as opposed to managing the state in the parent component. Moving the search functionality and state to this component should help clean up this pattern.